stdlib/public/core/Integers.swift.gyb

```diff
@@ -3885,41 +3885,29 @@ extension ${Self} : Hashable {
   ///   of this instance.
   @inlinable
   public func hash(into hasher: inout Hasher) {
-    // FIXME(hasher): To correctly bridge `Set`s/`Dictionary`s containing
-    // `AnyHashable`-boxed integers, all integer values are currently required
-    // to hash exactly the same way as the corresponding (U)Int64 value.  To fix
-    // this, we should introduce a custom AnyHashable box for integer values
-    // that sign-extends values to 64 bits.
-    % if bits <= word_bits:
-    hasher._combine(_lowWord)
-    % elif bits == 2 * word_bits:
-    if let word = ${"" if signed else "U"}Int(exactly: self) {
-      hasher._combine(word._lowWord)
-    } else {
-      hasher._combine(UInt64(_value))
-    }
-    % else:
-    fatalError("Unsupported integer width")
-    % end
+    hasher._combine(${U}${Self}(_value))
   }
 
   @inlinable
   public func _rawHashValue(seed: (UInt64, UInt64)) -> Int {
-    // FIXME(hasher): Note that the AnyHashable concern applies here too,
-    // because hashValue uses top-level hashing.
-    % if bits <= word_bits:
-    return Hasher._hash(seed: seed, _lowWord)
-    % elif bits == 2 * word_bits:
-    if let word = ${"" if signed else "U"}Int(exactly: self) {
-      return Hasher._hash(seed: seed, word._lowWord)
-    }
-    return Hasher._hash(seed: seed, UInt64(_value))
+    % if bits < word_bits:
+    return Hasher._hash(
+      seed: seed,
+      bytes: UInt64(truncatingIfNeeded: ${U}${Self}(_value)),
+      count: ${bits / 8})
     % else:
-    fatalError("Unsupported integer width")
+    return Hasher._hash(seed: seed, ${U}${Self}(_value))
     % end
   }
 }
 
+extension ${Self} : _HasCustomAnyHashableRepresentation {
+  // Not @inlinable - blacklisted
```

Comment:
What does this mean? (Also, we try to avoid the word "blacklist" both for being unspecific and for potentially being culturally charged.)